### PR TITLE
fix: expose multimodal agent metrics

### LINKED ISSUE #1010: Expose metrics from MultimodalAgent and RealtimeModel

### PR DESCRIPTION
**Pull request by longcw**
closes #1010

### CODE REVIEW

File: livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/realtime/realtime_model.py [hunk 1]
```diff
@@ -66,6 +69,10 @@ class RealtimeResponse:
     """usage of the response"""
     done_fut: asyncio.Future[None]
     """future that will be set when the response is completed"""
+    created_timestamp: float
+    """timestamp when the response was created"""
+    first_token_timestamp: float | None = None
+    """timestamp when the first token was received"""
```
Comment (theomonnom) [inline livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/realtime/realtime_model.py:75]: We should keep those states private

File: livekit-agents/livekit/agents/metrics/base.py [hunk 2]
```diff
@@ -98,6 +98,31 @@ class PipelineVADMetrics(VADMetrics):
...
+        audio_tokens: int
+
+    @dataclass
+    class OutputTokenDetails:
+        text_tokens: int
+        audio_tokens: int
+
+    input_token_details: InputTokenDetails
+    output_token_details: OutputTokenDetails
+
```
Comment (theomonnom) [inline livekit-agents/livekit/agents/metrics/base.py:124]: Let's reuse LLMMetrics and add more fields to it

Comment (longcw) [reply livekit-agents/livekit/agents/metrics/base.py:124]: So did you mean to add `input_token_details` and `output_token_details` to `LLMMetrics`?

Comment (theomonnom) [reply livekit-agents/livekit/agents/metrics/base.py:124]: No, feel free to ignore, I thing creating a new dataclass is fine

File: livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/realtime/realtime_model.py (hunk 1 above)
Comment (longcw) [reply livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/realtime/realtime_model.py]: Renamed to `_created_timestamp` and `_first_token_timestamp`.

File: livekit-agents/livekit/agents/metrics/base.py [hunk 3]
```diff
@@ -98,6 +98,31 @@ class PipelineVADMetrics(VADMetrics):
...
+@dataclass
+class MultiModalLLMError(Error):
+    type: str | None
+    reason: str | None = None
+    code: str | None = None
+    message: str | None = None
+
+
+@dataclass
+class MultiModalLLMMetrics(LLMMetrics):
```
Comment (theomonnom) [inline livekit-agents/livekit/agents/metrics/base.py:110]: For consistency with MultimodalAgent
```suggestion
class MultimodalLLMMetrics(LLMMetrics):
```

File: livekit-agents/livekit/agents/metrics/base.py [hunk 4]
```diff
@@ -98,6 +98,31 @@ class PipelineVADMetrics(VADMetrics):
     pass
 
 
+@dataclass
+class MultiModalLLMError(Error):
```
Comment (theomonnom) [inline livekit-agents/livekit/agents/metrics/base.py:102]:
```suggestion
class MultimodalLLMError(Error):
```

**Review by theomonnom — APPROVED** [review]

**Pull request merged**